test/index.js

```diff
@@ -547,6 +547,30 @@ describe('Nightmare', function () {
       value.should.equal('');
     })
 
+    it('should not type in a nonexistent selector', function*(){
+      var failed = false;
+      try {
+        yield nightmare
+          .goto(fixture('manipulation'))
+          .type('does-not-exist', 'nightmare');
+      } catch(e){
+        failed = true;
+      }
+      failed.should.be.true;
```

Comment:
It is probably out of scope, but year, I was mainly talking about just these rejection tests. TBH, I was thinking about adding it in the `goto` PR, but felt like it was out of scope there, too. Maybe better as its own separate PR  :P

Mainly it’s just that there should be less rigamarole and better error messaging than this pattern gives us and `mocha-generators` doesn’t seem to patch `should.Throw`, which would be the right generator-based way to do this.